packages/TCSLib/TCSLib.0.3/opam

```diff
@@ -0,0 +1,30 @@
+opam-version: "1.2"
+name: "TCSLib"
+version: "0.3"
+maintainer: "opam-devel@lists.ocaml.org"
+authors: [ "Oliver Friedmann"
+           "Martin Lange" ]
+license: "BSD-3-clause"
+homepage: "https://github.com/tcsprojects/tcslib"
+dev-repo: "git://github.com/tcsprojects/tcslib.git"
+bug-reports: "https://github.com/tcsprojects/tcslib/issues"
+build: [
+  ["ocaml" "setup.ml" "-configure" "--prefix" prefix]
+  ["ocaml" "setup.ml" "-build"]
+]
+install: ["ocaml" "setup.ml" "-install"]
+remove: [
+  ["ocamlfind" "remove" "TCSLib"]
+]
+build-test: [
+  ["ocaml" "setup.ml" "-configure" "--enable-tests"]
+  ["ocaml" "setup.ml" "-build"]
+  ["ocaml" "setup.ml" "-test"]
+]
+depends: [
+  ("extlib" | "extlib-compat")
+  "ocamlbuild" {build}
+  "ocamlfind" { <= "1.7.1" }
+  "num"
+]
+available: [ ocaml-version >= "4.06.0" ]
```

Comment:
Why is there such a constraint ? After modifying the _oasis file I could compile it with 4.05.0 without any problems.

Reply:
I was essentially trying to make the "num" dependency work as it seems to have been outsourced from OCaml in version 4.06.0.

There also seems to be an issue with ocamlfind 1.7.3 to install num:

https://github.com/ocaml/opam-repository/issues/10925

And it was suggested there to use ocamlfind 1.7.1

It did not get any of these variants to work so far - I'd be happy to remove any of these restrictions as long as it will also be working with the newest version of OCaml.

Reply:
For the ```num``` dependency it really doesn't matter as long as you use the num package correctly; and it seems you do as you seem to use the ```num``` package from ocamlfind correctly.

Regarding the ocamlfind issue it seems to have been closed by the author and therefor it is unlikely to be a wildspread bug (maybe it's an opam bug of some kind).

As long as the CI is happy this should be fine. I think those restrictions can be removed safely.

Reply:
So you'd recommend to remove the explicit ocamlfind dependency and e.g. set ocaml-version >= "4.03.0"?

Thanks,
Oliver

Reply:
Yes, I think it should be set to the lowest ocaml version possible as long as it compiles and works (which come in pair 99% of the time)